llama/llama.go

```diff
@@ -128,11 +128,21 @@ func (c *Context) SampleTokenGreedy(logits []float32) int {
 		ptr.p = 0.0
 	}
 
-	return int(C.llama_sample_token_greedy(c.c, &C.llama_token_data_array{
+	// Since the logits are provided externally, we unbox the
+	// llama_sampler_sample implementation here
+	// CITE: https://github.com/ggerganov/llama.cpp/blob/master/include/llama.h#L1154
+	sampler := C.llama_sampler_init_greedy()
+	cur_p := &C.llama_token_data_array{
```

Comment:
Thanks a lot for updating this function but I think it's best to just remove SampleTokenGreedy. In general, the goal is to keep the CGo layer as small and thin as possible. Given that this is getting more complex and is not used, there's really no need to carry it.

Reply:
Makes sense! I'll remove it